### ISSUE TITLE
Rule Change: (Add destructuredArrayIgnorePattern option to no-unused-vars)

### ISSUE THREAD

**Issue by JJPell** [eslint/eslint#15611]
### What rule do you want to change?

no-unused-vars

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// With the following option set within eslintrc rules
// "no-unused-vars": ["error", { "destructuredArrayIgnorePattern": "^_" }],

const array = ['a', 'b', 'c'];

const [a, _b, c] = array;

const newArray = [a, c];

// No "no-unused-vars" error on _b
```

### What does the rule currently do for this code?

Currently `_b` would be seen as an unused variable.

### What will the rule do after it's changed?

The proposed rule would allow a pattern to be set to ignore certain deconstructed array variables from triggering `no-unused-vars`.

It would essentially work in the same way as the existing `argsIgnorePattern` option for `no-unused-vars`.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Currently it is possible to avoid the `no-unused-vars` error by changing the example code to the following:

```javascript
const [a, , c] = array;
```

However in my opinion this doesn't read as well as either

```javascript
const [a, _b, c] = array;
```

or

```javascript
const [a, _, c] = array;
```

And so it would be great to have this option.

**Comment by mdjermanovic** [comment]
Makes sense to me :+1:

**Comment by nzakas** [comment]
No objections. @JJPell do you want to submit a pull request?

**Comment by snitin315** [comment]
Makes sense 👍🏻, I can submit a PR for this.

**Comment by nzakas** [comment]
@snitin315 lets see if @JJPell wants to first.

**Comment by JJPell** [comment]
Unfortunately I am unable to currently and so would welcome anybody who wishes to take this on.

**Comment by nzakas** [comment]
Okay @snitin315 you’re up.
